app/inject/ProductionModule.java

```diff
@@ -23,7 +25,9 @@
         return seq(
                 bind(CountryCode.class).qualifiedWith("default").toInstance(defaultCountry(configuration)), // checks on start
                 bind(PlayJavaSphereClient.class).toProvider(PlayJavaSphereClientProvider.class).in(Singleton.class),
-                bind(CategoryTree.class).toProvider(CategoryTreeProvider.class)
+                bind(CategoryTree.class).toProvider(CategoryTreeProvider.class),
```

Comment:
I think Providers are cool but there is the danger that you expect a singleton where it isn't. (at-Singleton on the provider reuses only the provider). Suppose you have multiple controllers, should everyone have its own copy of a CategoryTree, TemplateService and CmsService or should they share one instance?

`bind(Hello.class).toProvider(HelloProvider.class).in(Singleton.class) //make singleton even if class have not the annotation`